Source/src/Components/Component.h

```diff
@@ -1,58 +1,81 @@
 #pragma once
 
 #include "Utils/UUID.h"
-#include "Utils/JsonFormaterValue.h"
+#include "Utils/JsonFormatterValue.h"
 
-class GameObject;
-class ComponentCamera;
-class Program;
-
-class Component
+namespace Hachiko
 {
-public:
-	enum Type
-	{
-		NONE,
-		TRANSFORM,
-		MESH,
-		MATERIAL,
-		CAMERA,
-		DIRLIGHT,
-		POINTLIGHT,
-		SPOTLIGHT,
-		UNKNOWN
-	};
+    class GameObject;
+    class ComponentCamera;
+    class Program;
+
+    class Component
+    {
+    public:
+        enum class Type
+        {
+            NONE,
+            TRANSFORM,
+            MESH,
+            MATERIAL,
+            CAMERA,
+            DIRLIGHT,
+            POINTLIGHT,
+            SPOTLIGHT,
+            UNKNOWN
+        };
+
+        Component(const Type type, GameObject* container) :
+            game_object(container),
+            type(type) {}
+
+        virtual ~Component() = default;
+
+        virtual void Update() { }
+
+        virtual void OnTransformUpdated() {}
+
+        [[nodiscard]] Type GetType() const
+        {
+            return type;
+        }
 
-	Component(Type type, GameObject* container)
-		: type(type)
-		, game_object(container) {}
+        [[nodiscard]] UID GetID() const
+        {
+            return uid;
+        }
 
-	virtual ~Component() {};
+        [[nodiscard]] const GameObject* GetGameObject() const
+        {
+            return game_object;
+        }
 
-	virtual void Update() {};
-	virtual void OnTransformUpdated() {};
+        GameObject* GetGameObject()
+        {
+            return game_object;
+        }
 
-	inline Type GetType() const { return type; };
-    Hachiko::UID GetID() const { return uid; }
+        void SetGameObject(GameObject* container)
+        {
+            game_object = container;
+        }
 
-	const GameObject* GetGameObject() const { return game_object; }
```

Comment:
Having GameObject* const instead of const GameObject* would let user manipulate the GameObject but not change it's address. I think we could do that here.